src/Routing/Middleware/RoutingMiddleware.php

```diff
@@ -150,6 +151,11 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
                 $e->getMessage(),
```

Comment:
Why do we need to catch the new exception too in routing middleware? It will be handled by errorhandler middleware isn't it?

Reply:
Then we would need to adjust the tests, as I wrote above.
We could also do that.

Reply:
By raising the new exception we are risking breaking any application/plugin tests that directly use the current exception class. Is such a break worth the trouble?

Reply:
This is a minor release, and affects then only tests, easy to fix with the namespace change.
So I think if it makes it cleaner and more straightforward then yes.

Reply:
Another option is make `Routing\Exception\RedirectionException` an alias of `Http\Exception\RedirectException`.

Reply:
Actually, nothing breaks now the way it is now
The Routing part still catches those exceptions - and otherwise ErrorHandler jumps in.
So the current way is maybe the most pragmatic one forward.
And it can be cleaned up in 5.x.